pkg/datastore/kubernetes/kubernetes.go

```diff
@@ -87,6 +71,55 @@ func NewDatastore(thisClusterID string, stopCh <-chan struct{}) (datastore.Datas
 	}, nil
 }
 
+func getSubmarinerClientset(host string, k8sSpec *datastoreSpecification) (*submarinerClientset.Clientset, error) {
+	// We need to check whether we need the provided CA (for a secure connection)
+	// For an insecure connection, both cases are equivalent, but we’ll never get the X.509 error
+	// First, try without specifying anything (using the system trust store)
+	config, err := getSubmarinerRESTConfig(host, false, k8sSpec)
+	if err != nil {
+		return nil, err
+	}
+	clientset, err := submarinerClientset.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	// Check we can list endpoints
+	_, err = clientset.SubmarinerV1().Endpoints(k8sSpec.RemoteNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		if urlError, ok := err.(*url.Error); ok {
+			if _, ok := urlError.Unwrap().(x509.UnknownAuthorityError); ok {
+				// Certificate error, try with the trust chain
+				config, err = getSubmarinerRESTConfig(host, true, k8sSpec)
+				if err != nil {
+					return nil, err
+				}
+				return submarinerClientset.NewForConfig(config)
```

Comment:
Should we do the check on L87 again with the new clientset?

Reply:
Yes, I thought of that too overnight — the function should ensure the result is consistent, whatever path is taken.